### PR TITLE
Log mask sequence update

### PR DESCRIPTION
**Pull request by NoisomePossum**
### What does this PR do?
Adds a note to the mask sequence documentation to let customers know about the upcoming feature in this PR: https://github.com/DataDog/datadog-agent/pull/4563. The feature will be released for Agent version 6.17/7.17 so please don't merge until that version is released.

### Motivation
Documenting a feature enhancement for mask sequences.

### Preview link

https://docs-staging.datadoghq.com/michael.sha/mask-sequence-update/agent/logs/advanced_log_collection/

### CODE REVIEW

**Comment by NoisomePossum** [comment]
The PR has been merged so this is ready to be added as soon as version 6.17/7.17 are released.

File: content/en/agent/logs/advanced_log_collection.md [hunk 1]
```diff
@@ -104,6 +104,24 @@ logs:
         pattern: (?:4[0-9]{12}(?:[0-9]{3})?|[25][1-7][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})
 ```
 
+**Note**: As of Agent version 6.17, the `replace_placeholder` string can expand references to capture groups such as `$1`, `$2` and so forth. See the following example:
```
Comment (l0k0ms) [inline content/en/agent/logs/advanced_log_collection.md]:
```suggestion
**Note**: With Agent version 7.17+, the `replace_placeholder` string can expand references to capture groups such as `$1`, `$2` and so forth. See the following example:
```

File: content/en/agent/logs/advanced_log_collection.md [hunk 2]
```diff
@@ -104,6 +104,24 @@ logs:
...
+
+```
+pattern: (data:) values
+replace_placeholder: "$1 [masked_values]"
+
+Resulting string: "data: [masked_values]"
+```
+
+If you want a string to follow the capture group with no space in between, use the format `${<group_number>}`:
```
Comment (l0k0ms) [inline content/en/agent/logs/advanced_log_collection.md]:
```suggestion
If you want a string to follow the capture group with no space in between, use the format `${<GROUP_NUMBER>}`:
```

**Comment by NBParis** [comment]
@NoisomePossum @sarina-dd good to go in prod. Let's merge as the Agent is soon to be released and the version is specified in the documentation

**Pull request merged**